### ISSUE TITLE
Support Range Header streaming

### ISSUE THREAD

**Issue by der-On** [stephank/hyper-staticfile#9]
For media files a lot of browsers are using the Range header to retrieve only parts of bigger files.

It would be nice to have support for this in hyper-staticfile. If it is out of scope of this project I would like to be pointed to a crate handling this kind of requests to media files/streaming.

Thank you.

**Comment by stephank** [comment]
I don't think it is outside the scope, but I've also not yet in investigated how much work this is. :)

**Comment by stephank** [comment]
I think we can do the evaluation in `FileResponseBuilder` too? Otherwise, sounds about right!

There's apparently even a crate for parsing the `Range` header: https://github.com/bancek/rust-http-range

**Comment by infinityb** [comment]
Hello, I have an interest in this feature and have been working on an implementation (referenced above).  It isn't quite where I want it to be yet but I expect to PR it in the next while.

Regards,

**Comment by der-On** [comment]
Much :heart: for this! Thank you.
